preview/MsixCore/MsixCoreInstaller/RegistryDevirtualizer.cpp

```diff
@@ -68,6 +68,54 @@ HRESULT RegistryDevirtualizer::Run(_In_ bool remove)
     return S_OK;
 }
 
+HRESULT RegistryDevirtualizer::DeleteKeyIfPresent(HKEY rootHive, std::wstring subKeyPath, std::wstring extensionName)
+{
+    if (!m_hiveFileNameExists)
+    {
+        return S_OK;
+    }
+    std::wstring rootPath = m_loadedHiveKeyName + L"\\Registry";
+    RETURN_IF_FAILED(m_rootKey.Open(HKEY_USERS, rootPath.c_str(), KEY_READ)); // root key until registry
+
+    // map real root hive to virtual root hive
+    PCWSTR virtualKeyPath = L"";
+    for (auto mapping : mappings)
+    {
+        if (mapping.actualRootHive == rootHive && mapping.actualSubkey == subKeyPath)
+        {
+            virtualKeyPath = mapping.virtualKey;
+        }
+    }
+
+    // Per user mapping
+    RegistryKey userClassesKey;
+    HRESULT hrOpenUserClassesKey = m_rootKey.OpenSubKey(virtualKeyPath, KEY_READ, &userClassesKey);
+    if (SUCCEEDED(hrOpenUserClassesKey))
+    {
+        RegistryKey extensionKey;
+        HRESULT hrUserExtensionKey = userClassesKey.OpenSubKey(extensionName.c_str(), KEY_READ, &extensionKey);
+        if (SUCCEEDED(hrUserExtensionKey))
+        {
+            HRESULT deleteUserKey = userClassesKey.DeleteSubKey(extensionName.c_str());
+        }
+    }
+
+    //Per machine key
+    RegistryKey machineClassesKey;
+    HRESULT hrOpenMachineClassesKey = m_rootKey.OpenSubKey(L"MACHINE\\Software\\Classes", KEY_READ, &machineClassesKey);

```

Comment:
>Software\\Classes [](start = 70, length = 17)

this is hardcoded for Software\\Classes, so it works for FTA, but i'd want this to be more generic, so that it could work for uninstall key.

Reply:
Modified to remove uninstall key passed from ARP handler as well. Appended subpath passed to end of the generated key so that it is more generic.